src/test/java/org/gridsuite/modification/server/modifications/tabularmodifications/TabularVoltageLevelModificationsTest.java

```diff
@@ -0,0 +1,88 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author AJELLAL Ali <ali.ajellal@rte-france.com>
+ */
+@Tag("IntegrationTest")
+public class TabularVoltageLevelModificationsTest extends AbstractNetworkModificationTest {
+    @Override
+    protected Network createNetwork(UUID networkUuid) {
+        return NetworkCreation.create(networkUuid, true);
+    }
+
+    @Override
+    protected ModificationInfos buildModification() {
+        List<ModificationInfos> modifications = List.of(
+                VoltageLevelModificationInfos.builder().equipmentId("v1").nominalVoltage(new AttributeModification<>(300., OperationType.SET)).highVoltageLimit(new AttributeModification<>(400., OperationType.SET)).lowVoltageLimit(new AttributeModification<>(299., OperationType.SET)).build(),
+                VoltageLevelModificationInfos.builder().equipmentId("v2").nominalVoltage(new AttributeModification<>(300., OperationType.SET)).highVoltageLimit(new AttributeModification<>(400., OperationType.SET)).lowVoltageLimit(new AttributeModification<>(299., OperationType.SET)).build()
+        );
+        return TabularModificationInfos.builder()
+                .modificationType("VOLTAGE_LEVEL_MODIFICATION")
+                .modifications(modifications)
+                .stashed(false)
+                .build();
+    }
+
+    @Override
+    protected ModificationInfos buildModificationUpdate() {
+        List<ModificationInfos> modifications = List.of(
+                VoltageLevelModificationInfos.builder().equipmentId("v1").nominalVoltage(new AttributeModification<>(500., OperationType.SET)).highVoltageLimit(new AttributeModification<>(502., OperationType.SET)).lowVoltageLimit(new AttributeModification<>(499., OperationType.SET)).build(),
+                VoltageLevelModificationInfos.builder().equipmentId("v2").nominalVoltage(new AttributeModification<>(500., OperationType.SET)).highVoltageLimit(new AttributeModification<>(502., OperationType.SET)).lowVoltageLimit(new AttributeModification<>(499., OperationType.SET)).build()
+        );
+        return TabularModificationInfos.builder()
+                .modificationType("VOLTAGE_LEVEL_MODIFICATION")
+                .modifications(modifications)
+                .stashed(false)
+                .build();
+    }
+
+    @Override
+    protected void assertAfterNetworkModificationCreation() {
+        assertEquals(300., getNetwork().getVoltageLevel("v1").getNominalV(), 0.001);
+        assertEquals(300., getNetwork().getVoltageLevel("v2").getNominalV(), 0.001);
+    }
+
+    @Override
+    protected void assertAfterNetworkModificationDeletion() {
+        assertEquals(380.0, getNetwork().getVoltageLevel("v1").getNominalV(), 0.001);
+        assertEquals(225.0, getNetwork().getVoltageLevel("v2").getNominalV(), 0.001);
```

Comment:
You can test also low/ligh voltage limit

Reply:
No need to check  low/high voltage limit here.

Reply:
it is important to check we've got the original values back, once the modification is no more applied.
that's what you do when you test "225" for v2 nominalVoltage.

Reply:
I didn't  check low/high voltage limit  because they are not set while creating the voltage levels 'v1' and 'v2' however  nominalVoltage is set.
So testing that nominalVoltage is equals to "225"/"380" is sufficient to verify that the modification is no more applied.

Reply:
ok low/high values are Nan in the original network. But is it nice to check it back here